app/src/main/java/com/wizeline/heroes/NetworkClient.kt

```diff
@@ -1,14 +1,26 @@
 package com.wizeline.heroes
 
 import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
 import retrofit2.converter.gson.GsonConverterFactory
 
-class NetworkClient {
+object NetworkClient {
+
+    private var retrofitInstance: Retrofit? = null
+
+    init {
+        if (retrofitInstance == null) createRetrofitInstance()
+    }
 
     fun getServices(): HeroesServices {
-        return Retrofit.Builder().apply {
+        return retrofitInstance!!.create(HeroesServices::class.java)
```

Comment:
We should avoid the operator `!!` because it can throw a runtime exception.
```suggestion
       if (retrofitInstance == null) createRetrofitInstance()
       return retrofitInstance.create(HeroesServices::class.java)
```

Reply:
Done. created a separate method that creates a new Retrofit instance if the global instance is null and return it. Therefore, it is safe to return the nullable instance as ```retrofitInstance!!```